src/browser/modules/Editor/Monaco.tsx

```diff
@@ -76,410 +65,380 @@ const MonacoStyleWrapper = styled.div`
-       * are located, into the overflowing content widgets div.
-       * This solves the command palette being squashed when the cypher editor is only a few lines high.
-       * The workaround is based on a suggestion found in the github issue: https://github.com/microsoft/monaco-editor/issues/70
-       */
-      const quickInputDOMNode = editorRef.current.getContribution<
-        { widget: { domNode: HTMLElement } } & editor.IEditorContribution
-      >('editor.controller.quickInput').widget.domNode
-      ;(editorRef.current as any)._modelData.view._contentWidgets.overflowingContentWidgetsDomNode.domNode.appendChild(
-        quickInputDOMNode.parentNode?.removeChild(quickInputDOMNode)
-      )
-      QuickInputList.prototype.layout = function(maxHeight: number) {
-        this.list.getHTMLElement().style.maxHeight =
-          maxHeight < 200 ? '200px' : Math.floor(maxHeight) + 'px'
-        this.list.layout()
-      }
+  private getMonacoId = (): string => `monaco-${this.props.id}`
+  private debouncedUpdateCode = debounce(() => {
+    const text =
+      this.editor
+        ?.getModel()
+        ?.getLinesContent()
+        .join('\n') || ''
+
+    this.props.onChange(text)
+    this.addWarnings(parse(text).referencesListener.queriesAndCommands)
+  }, 300)
+  focus = (): void => {
+    this.editor?.focus()
+  }
+  setPosition = (pos: { lineNumber: number; column: number }): void => {
+    this.editor?.setPosition(pos)
+  }
+  private newLine = (): void =>
+    this.editor?.trigger('keyboard', 'type', { text: '\n' })
+  private isMultiLine = (): boolean =>
+    (this.editor?.getModel()?.getLineCount() || 0) > 1
+
+  private updateGutterCharWidth = (dbName: string): void => {
+    this.editor?.updateOptions({
+      lineNumbersMinChars:
+        dbName.length && !this.isMultiLine() ? dbName.length * 1.3 : 2
+    })
+  }
 
-      return () => {
-        editorRef.current?.dispose()
-        debouncedUpdateCode.cancel()
-      }
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [])
+  resize = (fillContainer: boolean): void => {
+    if (!this.container || !this.editor) return
+    const contentHeight = this.editor.getContentHeight()
 
-    const isFullscreenRef = useRef<boolean>(false)
+    const height = fillContainer
+      ? Math.min(window.innerHeight - 20, this.container.scrollHeight)
+      : Math.min(276, contentHeight) // Upper bound is 12 lines * 23px line height = 276px
 
-    const resize = (fillContainer: boolean) => {
-      const container = document.getElementById(monacoId) as HTMLElement
-      const contentHeight = editorRef.current?.getContentHeight() || 0
+    this.container.style.height = `${height}px`
+    this.editor.layout({
+      height,
+      width: this.container.offsetWidth
+    })
+  }
 
-      isFullscreenRef.current = fillContainer
+  getValue = (): string => this.editor?.getValue() || ''
+  setValue = (value: string): void => {
+    this.setState({ currentHistoryIndex: -1 })
```

Comment:
Kind of makes me think we should make -1 a constant, something like `UNSET_HISTORY`